src/api-service.jsx

```diff
@@ -0,0 +1,74 @@
+const baseUrl = 'https://webdev-hw-api.vercel.app/api/v1/prod/instapro';
+
+export function renderAllPosts(setAllPosts) {
+    return fetch('https://webdev-hw-api.vercel.app/api/v1/prod/instapro')
+        .then((response) => response.json())
+        .then((data) => setAllPosts(data.posts))
+        .then((data) => console.log(data))
+}
+
+export function renderAllUsersPosts(setAllPosts, userId) {
+    fetch(`https://webdev-hw-api.vercel.app/api/v1/prod/instapro/user-posts/${userId}`)
+        .then((response) => response.json())
+        .then((data) => setAllPosts(data.posts))
+}
+
+export function loginClick(login, password, setUserToken, navigate) {
+    if (login.length && password.length) {
+        fetch('https://webdev-hw-api.vercel.app/api/user/login', {

```

Comment:
чтобы работать с данными снаружи, нужно промис из этой функции вернуть, тогда ты с помощью then или await сможет из него данные получить:

```suggestion
        return fetch('https://webdev-hw-api.vercel.app/api/user/login', {

```

Reply:
Здесь и в остальных местах